chatmd/util.py

```
@@ -94,16 +97,69 @@ def set_cached_folder_id(name: str, folder_id: str) -> None:
+            except OSError:
+                pass
+        return None
+
+    diff_path.write_text(output, encoding="utf-8")
+    return diff_path
+
+
+def snapshot_for_diff(path: Path) -> Optional[Path]:
+    if not path.exists():
+        return None
+    tmp = tempfile.NamedTemporaryFile(delete=False, suffix=path.suffix)
```

Ruff (0.15.9)

[warning] Use a context manager for opening files

(SIM115)